src/web/app/pages-admin/admin-search-page/admin-search-page.component.ts

```diff
@@ -189,4 +192,32 @@ export class AdminSearchPageComponent {
     return link.replace(regex, `$1${newVal}`);
   }
 
+  /**
+   * Open up an email populated with content for course join invitation.
+   */
+  openCourseJoinEmail(courseId: string, studentemail: string): void {
+    this.emailGenerationService.getCourseJoinEmail(courseId, studentemail)
+        .subscribe((email: Email) => {
+          window.location.href = `mailto:${email.recipient}`
+              + `?Subject=${email.subject}`
+              + `&body=${email.content}`;
+        }, (err: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorToast(err.error.message);
+        });
+  }
+
+  /**
+   * Open up an email populated with content for feedback session reminder.
+   */
+  openFeedbackSessionReminderEmail(courseId: string, studentemail: string, fsname: string): void {
+    this.emailGenerationService.getFeedbackSessionReminderEmail(courseId, studentemail, fsname)
+        .subscribe((email: Email) => {
+          window.location.href = `mailto:${email.recipient}`
+              + `?Subject=${email.subject}`
+              + `&body=${email.content}`;
+        }, (err: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorToast(err.error.message);
+        });
```

Comment:
The body of these two methods are identical other than the method being called in the service class. Can they be abstracted out?

Reply:
Don't think it's a good idea to refactor this. The function parameters could always change to require even more arguments, and these methods might not share similar parameters at all. The only repeated code I see here is this portion
```
window.location.href = `mailto:${email.recipient}`
              + `?Subject=${email.subject}`
              + `&body=${email.content}`;
```
where it is already very self descriptive. Could be worth refactoring if we further add similar methods?

Reply:
What I'm suggesting is a function that takes in `Observable<Email>`, so it shouldn't matter how many parameter it takes or whether the parameters change or not.
Since this is repeated just twice, it is fine to let this be for now.